### PR TITLE
[feat] : Created bug report form in yaml. #12214

### PR DESCRIPTION
**Pull request by omsurase**
## Purpose / Description
Bug report form was not present. as talked in issue #12214

## Fixes
created bug report form in yaml.
file location: main/.github/ISSUE_TEMPLATE/bugreportform.yaml

## Approach
Created bug report form using yaml. Used yaml docs and existing issue report template in markdown as reference.

## How Has This Been Tested?
I have tested it by reporting bug in my repository.

## Learning (optional, can help others)
Not Applicable

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
Thanks!!!

Could you squash the commits into one.

Could you post a link to the repository/created issue to see what things look like

File: .github/ISSUE_TEMPLATE/bugreportform.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to fill out this bug/issue report!
+        
+  - type: checkboxes
+    id: contact
+    attributes:
+      label: Is there an existing issue for this?
+      description: Please go through open issues, to see if anything similar already exists.
```
Comment (david-allison) [inline .github/ISSUE_TEMPLATE/bugreportform.yaml:16]: Should have a link

File: .github/ISSUE_TEMPLATE/bugreportform.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      required: true 
+      
+  - type: textarea
+    id: debug-info
+    attributes:
+      label: Debug info
+      description: Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".
+      placeholder: Debug info.
+    validations:
+      required: true
```
Comment (david-allison) [inline .github/ISSUE_TEMPLATE/bugreportform.yaml:56]: What options do we have for validation? Can we ensure that this is filled in with the correct content via a quick regex?

File: .github/ISSUE_TEMPLATE/bugreportform.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      label: Research
+      description: Confirm the points below
+      options:
+        - label: I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
+          required: true
+        - label: I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue  
+          required: true
+        - label: I have searched for similar existing issues here and on the user forum
+          required: true
+        - label: I have confirmed the issue is not resolved in the latest alpha release [instructions](https://docs.ankidroid.org/manual.html#betaTesting)
```
Comment (david-allison) [inline .github/ISSUE_TEMPLATE/bugreportform.yaml:76]: Most people don't fill these in. They shouldn't be required

**Comment by omsurase** [comment]
link for test bug report. : https://github.com/omsurase/Anki-Android/issues/1
link for my repository. : https://github.com/omsurase/Anki-Android

you could try out this form by creating a new issue in my repository.
@david-allison

File: .github/ISSUE_TEMPLATE/bugreportform.yaml (hunk 1 above)
Comment (omsurase) [reply .github/ISSUE_TEMPLATE/bugreportform.yaml:16]: could you explain this in more detail.

File: .github/ISSUE_TEMPLATE/bugreportform.yaml (hunk 3 above)
Comment (omsurase) [reply .github/ISSUE_TEMPLATE/bugreportform.yaml:76]: should i make all of these as optional?

File: .github/ISSUE_TEMPLATE/bugreportform.yaml (hunk 1 above)
Comment (david-allison) [reply .github/ISSUE_TEMPLATE/bugreportform.yaml:16]: This should link the open issues so a user has less to do

File: .github/ISSUE_TEMPLATE/bugreportform.yaml (hunk 3 above)
Comment (david-allison) [reply .github/ISSUE_TEMPLATE/bugreportform.yaml]: Unsure. `I have searched for similar existing issues here and on the user forum` doesn't seem necessary with the checkbox above

`I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid` - I don't think this provides value. I think what we're really asking is "is this AnkiDroid or Anki Desktop", which would be much easier to ask

`I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue  ` - probably required

`I have confirmed the issue is not resolved in the latest alpha release [instructions](https://docs.ankidroid.org/manual.html#betaTesting)` - optional

@mikehardy thoughts?

File: .github/ISSUE_TEMPLATE/bugreportform.yaml (hunk 2 above)
Comment (omsurase) [reply .github/ISSUE_TEMPLATE/bugreportform.yaml:56]: I don't think there is any way for validation of text that's written inside debug info's text box (I might be wrong, but didn't find any resources or examples regarding this.).
there are third-party tools that validate the syntax of YAML files, but I don't think that's helpful in the validation of content inside the textboxes.

**Comment by omsurase** [comment]
I am facing some problems with my git terminal, therefore can't merge the commits into a single commit.

File: .github/ISSUE_TEMPLATE/bugreportform.yaml (hunk 3 above)
Comment (mikehardy) [reply .github/ISSUE_TEMPLATE/bugreportform.yaml]: I agree on all that

**Comment by david-allison** [comment]
> I am facing some problems with my git terminal, therefore can't merge the commits into a single commit. ill resolve this ASAP.

`git rebase -i HEAD~60` should let you squash your commits.

Or select all the commits in Android Studio's git window and select `squash`

File: .github/ISSUE_TEMPLATE/bugreportform.yaml (hunk 2 above)
Comment (david-allison) [reply .github/ISSUE_TEMPLATE/bugreportform.yaml:56]: That's a shame. I've checked and agree with your analysis

**Comment by omsurase** [comment]
@david-allison  i have created a new pr [link](https://github.com/ankidroid/Anki-Android/pull/12248)
we can continue this there.

**Comment by david-allison** [comment]
Hi, can you aim to keep changes to one PR in future (force pushing if you need to). Thanks!

**Comment by omsurase** [comment]
thanks for the feedback :). I will keep changes restricted to 1 PR in the future, apologies for this inconvenience.

**Pull request closed without merging**